go/tasks/pluginmachinery/utils/marshal_utils.go

```diff
@@ -68,3 +68,21 @@ func MarshalObjToStruct(input interface{}) (*structpb.Struct, error) {
 	}
 	return structObj, nil
 }
+
+// Don't use this if the unmarshalled obj is a proto message.
+func UnmarshalStructToObj(structObj *structpb.Struct, obj interface{}) error {
```

Comment:
These are becoming confusing :D I think this naming is better than the other ones in the file... there are also a bunch of them here: https://github.com/flyteorg/flytestdlib/blob/519a271602eb6c88d58be3cf1576d03113beee31/utils/marshal_utils.go#L17

mind filing a housekeeping bug to clean that mess?

Reply:
sure, done https://github.com/flyteorg/flyte/issues/828